src/PerfView/EtwEventSource.cs

```diff
@@ -552,7 +554,12 @@ internal ETWEventRecord(ETWEventSource source, TraceEvent data, Dictionary<strin
 
                 m_timeStampRelativeMSec = data.TimeStampRelativeMSec;
                 m_idx = data.EventIndex;
+                m_payloads = new Payload[data.PayloadNames.Length];
```

Comment:
I think it would be good to change the type of `m_payloads` to a `List<Payload>` so that you can add new items to it after iterating over the fields in the event.  If you look at `ETWEventRecord..ctor`, there are a number of conditional calls to `AddField`.  It would be ideal to add these fields to `m_payloads` as well, so that what's in the single line view is also in the multi-line view.  Bonus points if the order matches.